backend/src/managers/UserManager.ts

```diff
@@ -284,9 +283,29 @@ export default class UserManager {
         return activeKarmaVotes;
     }
 
+    async removeVotesWhenKarmaIsLow(userId: number) {
+        // cached check status from redis
+        const cachedStatus = await this.redis.get(`remove_votes_when_karma_is_low_${userId}`);
+        if (cachedStatus) {
+            return;
+        }
+
+        const karmaVotes = await this.voteRepository.getVotesByUser(userId);
+        for (const vote of karmaVotes) {
+            await this.voteRepository.userSetVote(vote.userId, 0, vote.voterId);
+        }
+        await this.redis.set(`remove_votes_when_karma_is_low_${userId}`, 'true');
+        await this.redis.expire(`remove_votes_when_karma_is_low_${userId}`,
+            60 * 30 + Math.floor(Math.random() * 60 * 5) /* 30-35 minutes */);
+    }
+
     async getUserRestrictions(userId: number): Promise<UserRestrictions> {
         const effectiveKarma = await this.getUserEffectiveKarma(userId);
 
+        if (effectiveKarma < -10) {
```

Comment:
magic number, maybe worth extracting into a class constant or add a TODO for the future refactoring

Reply:
At this moment all “magic constants” are neatly contained inside a single method, if they were defined in a different place and used here (imo) it would only hurt the readability. But I agree that when (if) these constants start being used elsewhere, they should be moved into a centralized place or config.